rocketpy/Function.py

```diff
@@ -2030,3 +2118,101 @@ def differentiate(self, x, dx=1e-6):
         # h = (10)**-300
         # z = x + h*1j
         # return self(z).imag/h
+
+
+def funcify_method(*args, **kwargs):
+    """Decorator factory to wrap methods as Function objects and save them as cached
+    properties.
+
+    Parameters
+    ----------
+    *args : list
+        Positional arguments to be passed to rocketpy.Function.
+    **kwargs : dict
+        Keyword arguments to be passed to rocketpy.Function.
+
+    Returns
+    -------
+    decorator : function
+        Decorator function to wrap callables as Function objects.
+
+    Examples
+    --------
+    >>> class Test():
+    ...     @Funcify(inputs=['x'], outputs=['y'])
+    ...     def func(x):
```

Comment:
The implementation seems to raising an error when the method has arguments, as found by @MateusStano and I could reproduce it. Still searching for a fix. For methods that implicitly define a function the decorator worked just fine, _e.g._:

```
class A:
        def __init__(self) -> None:
            pass

        @funcify_method(inputs="t", outputs="t2")
        def square(self):
            return lambda t: t**2

    a = A()

    print(a.square(3))
```